### PR TITLE
pubsub: report SendAck errors through Receive

### LINKED ISSUE #695: internal/pubsub: deal with error returned from SendAcks
Currently the code does nothing in this case.
Should we:
* log
* panic
* increment a counter
* send it to Open Census
* something else
?

### PR DESCRIPTION
**Pull request by jba**
Fixes #695.

### CODE REVIEW

File: pubsub/pubsub.go [hunk 1]
```diff
@@ -159,15 +159,11 @@ type Subscription struct {
...
-	err   error
-	waitc chan struct{} // for goroutines waiting on ReceiveBatch
-
-	// cancel cancels all SendAcks calls.
-	cancel func()
+	mu     sync.Mutex
+	q      []*Message    // local queue of messages downloaded from server
+	err    error         // permanent error
+	waitc  chan struct{} // for goroutines waiting on ReceiveBatch
+	cancel func()        // for canceling all SendAcks calls.
```
Comment (ijt) [inline pubsub/pubsub.go]: Remove trailing dot.

File: pubsub/pubsub.go [hunk 2]
```diff
@@ -159,15 +159,11 @@ type Subscription struct {
...
-	mu sync.Mutex
-
-	// q is the local queue of messages downloaded from the server.
-	q     []*Message
-	err   error
-	waitc chan struct{} // for goroutines waiting on ReceiveBatch
-
-	// cancel cancels all SendAcks calls.
-	cancel func()
+	mu     sync.Mutex
```
Comment (ijt) [inline pubsub/pubsub.go]: Shouldn't mu have a comment telling what it protects?

Comment (jba) [reply pubsub/pubsub.go]: By convention, it protects everything underneath it until the next blank line. I'll add that though.

File: pubsub/acks_test.go [hunk 3]
```diff
@@ -297,3 +298,34 @@ func TestSubShutdownCanBeCanceledEvenWithHangingSendAcks(t *testing.T) {
...
+	sub := pubsub.NewSubscription(ds)
+	defer sub.Shutdown(ctx)
+	m, err := sub.Receive(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	m.Ack()
+	// Wait for the ack to be sent.
+	<-ackChan
+	// Wait a little longer for the logic after SendAcks returns to happen.
```
Comment (vangent) [inline pubsub/acks_test.go]: This seems potentially flaky... could it maybe be updated to call `Receive` until it gets an error?

**Pull request merged**